pkg/types/aws/validation/machinepool_test.go

```diff
@@ -250,3 +251,53 @@ func Test_validateAMIID(t *testing.T) {
 		})
 	}
 }
+
+func Test_validateCPUOptions(t *testing.T) {
+	cases := []struct {
+		platform *aws.Platform
+		pool     *aws.MachinePool
+
+		err string
+	}{{
+		pool: &aws.MachinePool{
+			CPUOptions: &aws.CPUOptions{
+				ConfidentialCompute: ptr.To(aws.ConfidentialComputePolicySEVSNP),
+			},
+		},
+	}, {
+		pool: &aws.MachinePool{
+			CPUOptions: &aws.CPUOptions{
+				ConfidentialCompute: ptr.To(aws.ConfidentialComputePolicyDisabled),
+			},
+		},
+	}, {
+		pool: &aws.MachinePool{
+			CPUOptions: &aws.CPUOptions{
+				ConfidentialCompute: ptr.To(aws.ConfidentialComputePolicy("")),
+			},
+		},
+		err: `^test-path.confidentialCompute: Invalid value: "": Allowed values are Disabled, AMDEncryptedVirtualizationNestedPaging and omitted$`,
+	}, {
+		pool: &aws.MachinePool{
+			CPUOptions: &aws.CPUOptions{
+				ConfidentialCompute: ptr.To(aws.ConfidentialComputePolicy("invalid")),
+			},
+		},
+		err: `^test-path.confidentialCompute: Invalid value: "invalid": Allowed values are Disabled, AMDEncryptedVirtualizationNestedPaging and omitted$`,
+	}, {
+		pool: &aws.MachinePool{
+			CPUOptions: &aws.CPUOptions{},
+		},
+		err: `^test-path.cpuOptions: Invalid value: "{}": At least one field must be set if cpuOptions is provided$`,
+	}}
+	for _, test := range cases {
+		t.Run("", func(t *testing.T) {
```

Comment:
For each test case, we should give it a name... In case, there is failure in the future, we can trace back which test is failing.